packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/AttributeStatement.cs

```diff
@@ -12,30 +11,63 @@ public sealed class AttributeStatement : MethodBodyStatement
     {
         public CSharpType Type { get; }
         public IReadOnlyList<ValueExpression> Arguments { get; }
+        public IReadOnlyDictionary<string, ValueExpression> PositionalArguments { get; }
 
-        public AttributeStatement(CSharpType type, IReadOnlyList<ValueExpression> arguments)
+        public AttributeStatement(CSharpType type, IReadOnlyList<ValueExpression> arguments, IReadOnlyDictionary<string, ValueExpression> positionalArguments)
         {
             Type = type;
             Arguments = arguments;
+            PositionalArguments = positionalArguments;
         }
 
-        public AttributeStatement(CSharpType type, params ValueExpression[] arguments) : this(type, (IReadOnlyList<ValueExpression>)arguments) { }
+        public AttributeStatement(CSharpType type, IReadOnlyList<ValueExpression> arguments) : this(type, arguments, new Dictionary<string, ValueExpression>()) { }
+
+        public AttributeStatement(CSharpType type, IReadOnlyDictionary<string, ValueExpression> positionalArguments) : this(type, [], positionalArguments) { }
+
+        public AttributeStatement(CSharpType type, params ValueExpression[] arguments) : this(type, arguments, new Dictionary<string, ValueExpression>()) { }
```

Comment:
I am not sure if this is the best practice, or we should use optional parameters